.github/workflows/main.yml

```diff
@@ -12,6 +12,10 @@ on:
     tags:
       - '*'
 
+concurrency:
+  group: ci-${{ github.head_ref }}
```

Comment:
we may need to adjust `head_ref` to `ref` for pushes to branches (Vs PRs)

Reply:
@runspired I'll merge but remove this for now, until you verify which one we need